### PR TITLE
[IMP] mass_mailing_themes: add new themes

### PR DESCRIPTION
**Pull request by anso-odoo**
5 new mail templates have been created :

2 designed templates ("Graphic" (now "Training") and "Coffeebreak")
3 text-heavy templates

XML added in mass_mailing_themes/views/mass_mailing_themes_templates.xml
Images added in mass_mailing_themes/static/src/img

Some templates names were adapted :

Newsletter -> Event
Tech -> Blogging
VIP -> Coupon
Bold -> Magazine
Solar -> Promotion

Co-Authored-By: Elisabeth Dickinson <edi@odoo.com>

Task-2714386

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
Hello,

I don't really manage mass mailing themes ^^ not sure who is really in charge of this module.

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 1]
```diff
@@ -6,13 +6,16 @@
             <div data-name="newsletter" title="Event Promo" data-img="/mass_mailing_themes/static/src/img/theme_imgs/newsletter_thumb" data-images-info='{"logo": {"format": "png"}, "all": {"module": "mass_mailing_themes"}}'>
                 <t t-call="mass_mailing_themes.theme_newsletter_template"/>
             </div>
+            <div data-name="textnews" title="Newsletter" data-layout-styles="background-color: rgb(255, 255, 255) !important;" data-img="/mass_mailing_themes/static/src/img/theme_imgs/malice_thumb" data-images-info='{"logo": {"format": "png"}, "all": {"module": "mass_mailing_themes"}}'>
```
Comment (dmo-odoo) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:9]: I'm a little bit confused here. The other themes seem to follow some naming convention such that "newsletter" for example has "newsletter_thumb". But this one has "malice_thumb" instead of "textnews_thumb". Is this intended ? Are you malicious ? 😈

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 2]
```diff
@@ -21,6 +24,18 @@
...
+                <t t-call="mass_mailing_themes.theme_coffeebreak_template"/>
+            </div>
+            <div data-name="roadshow1" title="Roadshow Schedule" data-layout-styles="background-color: rgb(255, 255, 255) !important;" data-img="/mass_mailing_themes/static/src/img/theme_imgs/roadshow_schedule_thumb" data-images-info='{"logo": {"format": "png"}, "all": {"module": "mass_mailing_themes"}}'>
+                <t t-call="mass_mailing_themes.theme_roadshow1_template"/>
+            </div>
+            <div data-name="roadshow2" title="Roadshow Follow-up" data-layout-styles="background-color: rgb(255, 255, 255) !important;" data-img="/mass_mailing_themes/static/src/img/theme_imgs/roadshow_followup_thumb" data-images-info='{"logo": {"format": "png"}, "all": {"module": "mass_mailing_themes"}}'>
+                <t t-call="mass_mailing_themes.theme_roadshow2_template"/>
+            </div>
+            <div data-name="graphic" title="Training" data-layout-styles="background-color: rgb(220, 234, 255) !important;" data-img="/mass_mailing_themes/static/src/img/theme_imgs/training_thumb" data-images-info='{"logo": {"format": "png"}, "all": {"module": "mass_mailing_themes"}}'>
```
Comment (dmo-odoo) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:36]: Same for the three above here. The name of the thumbnail does not seem to match the `data-name` of the theme. It's no big deal IMO but I just wanted to point it out in case it was a mistake or so. 🤷

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 3]
```diff
@@ -866,4 +881,750 @@
...
+                </div>
+                <div class="row">
+                    <div class="col-lg-12 o_mail_footer_links">
+                        <a role="button" href="/unsubscribe_from_list" class="btn btn-link text-o-color-4">Unsubscribe</a> | <a role="button" href="/contactus" class="btn btn-link text-o-color-4">Contact</a>
+                    </div>
+                </div>
+                <div class="row">
+                    <div class="col-lg-12">
+                        <p class="o_mail_footer_copy text-o-color-4">
+                            © <t t-esc="datetime.datetime.now().year"/> All Rights Reserved
```
Comment (dmo-odoo) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1052]: I'm curious because there is a t-set now at the top of every template but here we don't use it ?

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 4]
```diff
@@ -866,4 +881,750 @@
...
+                text-decoration-line: underline;
+                color: #212529;
+            }
+            a.btn.btn-primary, a.btn.btn-outline-primary, a.btn.btn-fill-primary {
+                background-color: #CE0000;
+                border-color: #CE0000;
+            }
+        </style>
+        <t t-set="now" t-value="datetime.datetime.now()"/>
+        <div class="o_snippet_view_in_browser o_mail_snippet_general pt16 pb0" style="text-align: left; padding-left: 15px; padding-right: 15px; color: rgb(183, 191, 199) !important" data-snippet="s_mail_block_header_view" data-name="View Online"><a href="/view" class="" target="_blank" data-bs-original-title="" title="">View Online</a> - <span style="color: rgb(183, 191, 199); font-size: 12.25px">August 29th 2022</span></div>
```
Comment (dmo-odoo) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1087]: Same here, do we really want to hardcode the date ? It's fine by me if that's what we want but it looks like a mistake.

**Comment by dmo-odoo** [comment]
Looks like there are issues with some styles in this PR regarding BS5, see here: https://runbot.odoo.com/runbot/build/18891788

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml (hunk 4 above)
Comment (anso-odoo) [reply addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml]: Date is now dynamic

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml (hunk 3 above)
Comment (anso-odoo) [reply addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1052]: I tried to remove them but it created errors :/

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml (hunk 2 above)
Comment (anso-odoo) [reply addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml]: I fixed all the template names to make everything more coherent 👍

Comment (mart-e) [reply addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml]: @anso-odoo no, no need to modify the PO files, that's the job of Transifex. As you are working in master, even updating the .pot is not required (but possible).

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml (hunk 3 above)
Comment (Zinston) [reply addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1053]: And you couldn't use it here ?

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 5]
```diff
@@ -1045,4 +1048,199 @@
...
+            a:not(.btn), a.btn.btn-link {
+                text-decoration-line: underline;
+                color: #212529;
+            }
+            a.btn.btn-primary, a.btn.btn-outline-primary, a.btn.btn-fill-primary {
+                background-color: #CE0000;
+                border-color: #CE0000;
+            }
+        </style>
+        <t t-set="now" t-value="datetime.datetime.now()"/>
```
Comment (Zinston) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1087]: This one is also not used. Why did you add it?

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 6]
```diff
@@ -1249,7 +1252,7 @@
             </div>
         </div>
     </template>
-    
+
```
Comment (Zinston) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1255]: Always nice :-)

**Review by Zinston — COMMENTED** [review]
Nothing shocked me, seems like good work :-) Only these `t-set`s to clean up, if they weren't there before and you're not using them, there's no reason you should have to keep them :p

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 7]
```diff
@@ -94,8 +94,11 @@
                                 <img src="/mass_mailing_themes/static/src/img/theme_event/s_default_image_media_list_1.jpg" class="s_media_list_img h-100 w-100" style="padding: 10px;"/>
                             </div>
                             <div class="col-lg-7 s_media_list_body">
-                                <h3>Get 6-months' worth of knowledge
-                                    <br/>in just 2 days</h3>
+                                <div class="s_title o_mail_snippet_general pb0 pt0" data-snippet="s_title" data-name="Title">
+                                    <div class="container s_allow_columns">
+                                        <h3>Get 6-months' worth of knowledge<br/>in just 2 days</h3>
+                                    </div>
+                                </div>
```
Comment (Zinston) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:101]: AFAIK, "Title" can't be dragged into another snippet. Couldn't this nested "Title" snippet cause issues? @qsm-odoo ?

I didn't find this spec in the task so I don't know what the goal was or what you mean with "add more dropouts". If it's just to have a "title" section in the sidebar, it's maybe a little strange imo since you can press enter at the end of your title and insert regular text:
image

**Comment by Zinston** [comment]
I took a look on Runbot, nice work! Just a little question about the last commit but other than that lgtm. I saw @LNAodoo ran some tests on emailonacid so I trust the HTML of the new templates gets properly converted?

FYI at first I couldn't see the whole musings template until I dropped a snippet in it but that was a pre-existing bug I wasn't aware of (so I went and fixed it: https://github.com/odoo/odoo/pull/125740).

File: addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml [hunk 8]
```diff
@@ -885,4 +909,744 @@
...
+            a.btn.btn-secondary, a.btn.btn-outline-secondary, a.btn.btn-fill-secondary {
+                color: #CE0000;
+                background-color: #FFFFFF;
+                border-color: #CE0000;
+            }
+        </style>
+        <div class="o_snippet_view_in_browser o_mail_snippet_general pt16 pb0" style="text-align: left; padding-left: 15px; padding-right: 15px; color: rgb(183, 191, 199) !important" data-snippet="s_mail_block_header_view" data-name="View Online"><a href="/view" class="" target="_blank" data-bs-original-title="" title="">View Online</a> - <span style="color: rgb(183, 191, 199); font-size: 12.25px" t-out="datetime.datetime.now().strftime('%B %d %Y')"></span></div>
+        <div class="s_title o_mail_snippet_general pt32 pb32" data-snippet="s_title" data-name="Title" style="background-color: rgb(255, 255, 255) !important;">
+            <div class="container s_allow_columns">
+                <h1 style="text-align:center" class="">Weekly Musing</h1>
```
Comment (Zinston) [inline addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1117]: Wouldn't you rather say "musings"? @edi-odoo ?

Comment (edi-odoo) [reply addons/mass_mailing_themes/views/mass_mailing_themes_templates.xml:1117]: Agreed @Zinston !

**Comment by KangOl** [comment]
Why doesn't this PR target `master`?

**Comment by LNAodoo** [comment]
> Why doesn't this PR target `master`?

Hello there,

We'd like to have this in stable partly because we need those on odoo.com for some ongoing newsletters.
And since we're "only" adding templates, I thought that would be OK? 

But maybe I'm missing something. Is there any change here that really cannot be done for 16.3? 

Thanks in advance for your help,
Cheers!

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

As it seems it was urgent, I rebased, removed useless images and pushed. If anything suspect appears, we will fix it after merge, as otherwise we would miss the 16.4 freeze :)

Cheers !
